### ISSUE TITLE
Keywords Outside of Configuration Section Cause Unexpected Behavior

### ISSUE THREAD

**Issue by scottschreckengaust** [PyCQA/pylint#2703]
### Steps to reproduce
1. Confirm a file has a pylint issue:
```bash
$ cat bug.py

if None:
     pass
$ pylint bug.py
************* Module bug
bug.py:3:0: W0311: Bad indentation. Found 5 spaces, expected 4 (bad-indentation)
bug.py:1:0: C0111: Missing module docstring (missing-docstring)
bug.py:2:0: W0125: Using a conditional statement with a constant value (using-constant-test)

-------------------------------------------------------------------
Your code has been rated at -5.00/10 (previous run: 0.00/10, -5.00)

```
2. Run pylint using a configuration file to see that the configuration file works:
```bash
$ cat works.ini
[MASTER]
ignore=bug.py
[MESSAGES CONTROL]
disable=missing-docstring,
        using-constant-test

$ pylint --rcfile=works.ini bug.py
```
3. Add a section in the configuration file for another tool that duplicates the key:
```bash
$ cat ignore-missed-by-pylint.ini
[MASTER]
ignore=bug.py
[MESSAGES CONTROL]
disable=missing-docstring,
        using-constant-test

[flake8]
ignore=E111

$ pylint --rcfile=ignore-missed-by-pylint.ini bug.py
************* Module bug
bug.py:3:0: W0311: Bad indentation. Found 5 spaces, expected 4 (bad-indentation)

--------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)

```

- This example has defined the `ignore` keyword for both pylint in the `[MASTER]` section and `[flake8]` section
- Changing the order of which section the `ignore` key is defined does not change the behavior
- Note that pylint is using the `disable` keyword properly, because it is listed only once.

### Current behavior
Duplicated keys, even if outside the documented section, may lead to unexpected behavior.

### Expected behavior
Only the key defined within the documented section is used.  If that key is also in an outside section, it should not be used.

### pylint --version output
```
pylint 2.2.2
astroid 2.1.0
Python 3.6.1 (default, Jul 27 2018, 09:38:45)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce the bug. For now it's safe to assume that adding new sections that are not `pylint` related it's most likely going to cause inadvertent side effects.
